src/net.c

```diff
@@ -2916,6 +2916,9 @@ handle_request(coap_context_t *context, coap_session_t *session, coap_pdu_t *pdu
                                      coap_encode_var_safe(buf, sizeof (buf),
                                                           resource->observe),
                                      buf);
+          } else {
+            resp = 409;
```

Comment:
This is an invalid response code for setting up an Observe registration.  As per [RFC7641 4.1. Request](https://datatracker.ietf.org/doc/html/rfc7641#section-4.1)
````
   A server that is unable or unwilling to add a new entry to the list
   of observers of a resource MAY silently ignore the registration
   request and process the GET request as usual.  The resulting response
   MUST NOT include an Observe Option, the absence of which signals to
   the client that it will not be notified of changes to the resource
   and, e.g., needs to poll the resource for its state instead.
````
which is how this should be processed.

Reply:
It is possible to use a 5.03 code here [RFC7252 Section 5.9.3.4 Service Unavailable](https://www.rfc-editor.org/rfc/rfc7252#section-5.9.3.4) (with a Max-Age option), but this then means that the client has to repeat the request without Observe option register set to get the current resource value.  I think that deleting lines 2919-2921 is the correct way to go as the client will not see the Observe option in the response and know that the Observe was not registered.